Added HAL function for popular bit-exact branch of GaussianBlur with sigma=0

asmorkalov opened the pull request:
Tries to address #24850
Previous trial: https://github.com/opencv/opencv/pull/25334

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

vpisarev commented on modules/imgproc/src/hal_replacement.hpp:
```diff
@@ -880,6 +880,25 @@ inline int hal_ni_gaussianBlur(const uchar* src_data, size_t src_step, uchar* ds
+   @param ksize Width of kernel
+   @param border_type Border type
+*/
+inline int hal_ni_gaussianBlurSymmetric(const uchar* src_data, size_t src_step, uchar* dst_data, size_t dst_step, int width, int height, int depth, int cn, size_t ksize, int border_type) { return CV_HAL_ERROR_NOT_IMPLEMENTED; }
```
1. ksize: `size_t => int` to match `width` and `height` types
2. add `int roi_x, int roi_y, int roi_width, int roi_height` to support ROI and external multi-threading
3. add `const uchar* border_value`

asmorkalov replied:
classic gaussianBlur uses size_t for kernel size too:
```
inline int hal_ni_gaussianBlur(const uchar* src_data, size_t src_step, uchar* dst_data, size_t dst_step, int width, int height, int depth, int cn, size_t margin_left, size_t margin_top, size_t margin_right, size_t margin_bottom, size_t ksize_width, size_t ksize_height, double sigmaX, double sigmaY, int border_type) { return CV_HAL_ERROR_NOT_IMPLEMENTED; }
```

asmorkalov replied:
border_value = scalarToRawData()

asmorkalov replied:
"add const uchar* border_value" - I'll do with dedicated PR. Need to change public API first.

vpisarev commented on modules/imgproc/src/hal_replacement.hpp:
```diff
@@ -880,6 +880,25 @@ inline int hal_ni_gaussianBlur(const uchar* src_data, size_t src_step, uchar* ds
+inline int hal_ni_gaussianBlurSymmetric(const uchar* src_data, size_t src_step, uchar* dst_data, size_t dst_step, int width, int height, int depth, int cn, size_t ksize, int border_type) { return CV_HAL_ERROR_NOT_IMPLEMENTED; }
+
+//! @cond IGNORED
+#define cv_hal_gaussianBlurSymmetric hal_ni_gaussianBlurSymmetric
```
also, why word "Symmetric" in the name? All Gaussian kernels are symmetric by definition

asmorkalov replied:
Let it be Binomial

asmorkalov commented:
@opencv-alalek @vpisarev I fixed remarks:
- Moved CALL_HAL after OpenCL.
- Renamed the function after vpisarev recommendation.
- Added margins as in classic GaussianBlur.
- presumed size_t and other types as in other HAL functions for consistency.
